relay-general/src/protocol/mechanism.rs

```diff
@@ -144,6 +144,43 @@ pub struct Mechanism {
     ///   as the user explicitly captured the exception (and therefore kind of handled it)
     pub handled: Annotated<bool>,
 
+    /// An optional string value describing the source of the exception.
+    ///
+    /// For chained exceptions, this should contain the platform-specific name of the property or
+    /// attribute (on the parent exception) that this exception was acquired from. In the case of
+    /// an array, it should include the zero-based array index as well.
+    ///
+    /// - Python Examples: `"__context__"`, `"__cause__"`, `"exceptions[0]"`, `"exceptions[1]"`
+    ///
+    /// - .NET Examples:  `"InnerException"`, `"InnerExceptions[0]"`, `"InnerExceptions[1]"`
+    ///
+    /// - JavaScript Examples: `"cause"`, `"errors[0]"`, `"errors[1]"`
+    #[metastructure(required = "false", nonempty = "true", max_chars = "enumlike")]
```

Comment:
Let's also limit the allowed characters for strings, see [example here](https://github.com/getsentry/relay/blob/3673a76f8efcb9827ce4e7d1fec60a36aab6c5ba/relay-general/src/protocol/tags.rs#L7).

Reply:
I added `deny_chars = " \t\r\n"` to prevent whitespace.  I don't want to specifically list allowable chars, because different languages have different conventions and it could be problematic to try to include all possible values.

Reply:
That makes sense. To prevent whitespace, it may be more readable to add `trim_whitespace` (and `nonempty`?), see [example](https://github.com/getsentry/relay/blob/ab2a3dca7a3daaecfcdf097d898c766fd6b69476/relay-general/src/protocol/event.rs#L383-L385).

Reply:
Trimming wouldn't catch whitespace in the middle of a string.